File: LC0128_Longest_Consecutive_Sequence/longest_consecutive_sequence.go
package main

func longestConsecutive(nums []int) int {
	// trivial cases
	if len(nums) == 0 {
		return 0
	}
	if len(nums) == 1 {
		return 1
	}

	// Create dictionary from nums- we really only care about the key (to check if a number exists in nums).
	nums_set := make(map[int]int)
	for _, v := range nums {
		nums_set[v] = 0
	}
	max_length := 1
	for num, _ := range nums_set {
		_, exists := nums_set[num-1]
		length := 1
		if !exists { // if num is the start of a sequence
			current := num

			for {
				_, exists := nums_set[current+1]
				if !exists {
					break
				}
				length++
				current++
			}
			if length > max_length {
				max_length = length
			}
		}
	}

	return max_length
}

File: LC0128_Longest_Consecutive_Sequence/longest_consecutive_sequence.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # trivial cases
        if len(nums) == 0:
            return 0
        elif len(nums) == 1:
            return 1

        max_length = 1
        nums = list(set(nums)) # Delete repeated numbers
        ranges = {num: [num, num] for num in nums} # maps number to the min and max of its longest consecutive elements sequence
        #print(ranges)
        for num in nums:
            # Check 1 below
            if num - 1 in ranges:
                ranges[num][0] = min(ranges[num][0], ranges[num - 1][0])
                ranges[ranges[num - 1][0]] = ranges[num]

            # Check 1 above
            if num + 1 in ranges:
                ranges[num][1] = max(ranges[num][1], ranges[num + 1][1])
                ranges[ranges[num + 1][1]] = ranges[num]

            max_length = max(max_length, ranges[num][1] - ranges[num][0] + 1)
        #print(ranges)
        return max_length